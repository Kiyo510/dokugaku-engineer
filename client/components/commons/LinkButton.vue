<template>
  <n-link :to="to" class="btn">
    <slot />
  </n-link>
</template>

<style lang="scss" scoped>
.btn {
  border: 1px solid transparent;
  border-radius: 6px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  padding: 1.2rem 2rem;
  text-align: center;
}

.btn-xs {
  font-size: $font-size-xs;
}

.btn-red1 {
  background-color: $color-red1;
  border-color: $color-red1;
  color: $color-white1;
}

.btn-outline-teal1 {
  border-color: $color-teal1;
  color: $color-teal1;
}
</style>

<script>
export default {
  props: {
    to: {
      type: String,
      default: '',
    },
  },
}
</script>
