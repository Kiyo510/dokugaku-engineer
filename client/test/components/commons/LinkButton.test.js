import {  mount, RouterLinkStub } from '@vue/test-utils'
import LinkButton from '~/components/commons/LinkButton.vue'

describe('components/commons/LinkButton.vue', () => {
  test('match snapshot', () => {
    const wrapper = mount(LinkButton, {
      stubs: {
        NLink: RouterLinkStub,
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
