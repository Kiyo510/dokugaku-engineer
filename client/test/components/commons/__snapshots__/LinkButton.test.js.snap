// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/LinkButton.vue match snapshot 1`] = `
<a
  class="btn"
/>
`;
